toolchain/check/testdata/interop/cpp/no_prelude/function_decl.carbon

```diff
@@ -176,6 +176,20 @@ fn F() {
   Cpp.foo();
 }
 
+// --- inline_function_decl.h
+
+inline void foo() {}
```

Comment:
Have you considered using a single .h file and giving entities distinct names? i.e., putting these inputs at the top of the file rather than intermingling them.

This is Dana's comment [on Discord](https://discord.com/channels/655572317891461132/655578254970716160/1368970112849285141)

Reply:
I would say that `function_decl.carbon` is doing far too much, and should be split up pretty heavily. Calling functions is going to be a pretty significant %age of the interop behaviour we have to test, it shouldn't be one test file.

Reply:
SGTM.
Moved the new tests to a new file and split the shard sets there with comments.
Will split this file further and add comments in a separate PR.